### PR TITLE
Prototype for monitor_dbus_signals test in stratisd_cert

### PR DESCRIPTION
**Pull request by bgurney-rh**
Supersedes: #70
Related: stratis-storage/project#369

### CODE REVIEW

**Review by mulkieran — CHANGES_REQUESTED** [review]

File: stratisd_cert.py [hunk 1]
```diff
@@ -200,6 +202,34 @@ def setUp(self):
...
+                shell=False,
+            )
+
+    def tearDown(self):
+        """
+        Tear down an individual test.  For now, this only stops the
+        D-Bus trace.
+        :return: None
+        """
+        try:
```
[the remark this answers is not in the document]
Comment (bgurney-rh) [reply stratisd_cert.py:224]: ...well, I think I found another reason to not just add the `time.sleep(1)` call in the try block: it resulted in none of the tests failing, when I previously had failures.

[the remark this answers is not in the document]
Comment (bgurney-rh) [reply stratisd_cert.py]: OK, now I see what's happening; those are spurious failures.

File: stratisd_cert.py [hunk 2]
```diff
@@ -200,6 +202,34 @@ def setUp(self):
         time.sleep(1)
         exec_command(["udevadm", "settle"])
 
+        # pylint: disable=consider-using-with
```
[the remark this answers is not in the document]
Comment (bgurney-rh) [reply stratisd_cert.py]: Resolved.

File: stratisd_cert.py [hunk 3]
```diff
@@ -200,6 +202,34 @@ def setUp(self):
...
+        Tear down an individual test.  For now, this only stops the
+        D-Bus trace.
+        :return: None
+        """
+        trace = getattr(self, "trace", None)
+        if trace is not None:
+            time.sleep(1)
+            self.trace.send_signal(signal.SIGINT)
+            (_, stderrdata) = self.trace.communicate()
+            self.trace.wait(timeout=1)
```
Comment (mulkieran) [inline stratisd_cert.py:229]: The wait command must be unnecessary. If communicate returned, the process has terminated. Note that communicate() takes a timeout argument as well, if you want one.

**Review by mulkieran — CHANGES_REQUESTED** [review]

File: testlib/utils.py [hunk 4]
```diff
@@ -76,6 +76,28 @@ def process_exists(name):
...
+    """
+    procs = []
+    for proc in psutil.process_iter(["cmdline"]):
+        try:
+            if any(name == param for param in proc.info["cmdline"]):
+                procs.append(proc)
+        except psutil.NoSuchProcess:
+            pass
+
+    if len(procs) > 0:
```
[the remark this answers is not in the document]
Comment (bgurney-rh) [reply testlib/utils.py]: OK; I confirmed that there were no stray test processes left behind.

File: testlib/utils.py [hunk 5]
```diff
@@ -76,6 +76,25 @@ def process_exists(name):
...
+    not in the process name.
+    :param name: name of script to clean up
+    :type name: str
+    return: None
+    """
+    procs = []
+    for proc in psutil.process_iter(["cmdline"]):
+        try:
+            if any(name == param for param in proc.info["cmdline"]):
+                for termproc in procs:
```
Comment (mulkieran) [inline testlib/utils.py:92]: Just get rid of the ```procs``` list entirely. The thing you want to call ```terminate()``` on is just the ```proc``` in ```for proc in psutil.process_iter(...):```.

**Review by mulkieran — CHANGES_REQUESTED** [review]
One last request, then I think we should be able to merge it since the code looks good and we're sure it works up to a point. After that, though, I think we'll want to stress it a bit and may find that we have to refine either this file or the monitor script.

File: testlib/utils.py (hunk 5 above)
Comment (bgurney-rh) [reply testlib/utils.py]: Resolved.

**Review by mulkieran — APPROVED** [review]
This looks ready to merge. Plz squash this PR or make a new PR as you see fit.

**Comment by mulkieran** [comment]
@bgurney-rh Plz close this one.

**Pull request closed without merging**